stores/mvstore/src/test/kotlin/kotlinx/document/database/tests/mvstore/MVStoreTests.kt

```diff
@@ -4,46 +4,94 @@ package kotlinx.document.database.tests.mvstore
 
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
+import kotlinx.document.database.DataStore
 import kotlinx.document.database.mvstore.MVDataStore
+import kotlinx.document.database.tests.AbstractCacheOverflowTests
 import kotlinx.document.database.tests.AbstractDeleteTests
 import kotlinx.document.database.tests.AbstractDocumentDatabaseTests
 import kotlinx.document.database.tests.AbstractFindTests
 import kotlinx.document.database.tests.AbstractIndexTests
 import kotlinx.document.database.tests.AbstractInsertTests
 import kotlinx.document.database.tests.AbstractObjectCollectionTests
+import kotlinx.document.database.tests.AbstractOnChangeCommitPolicyTests
+import kotlinx.document.database.tests.AbstractPeriodicCommitPolicyTests
 import kotlinx.document.database.tests.AbstractUpdateTests
 import kotlinx.document.database.tests.DatabaseDeleter
+import java.io.File
 import kotlin.io.path.Path
 import kotlin.io.path.deleteIfExists
 
+private val defaultTestConfiguration by lazy {
+    DataStore.Configuration(
+        kotlinx.io.files.Path(DB_PATH),
+        DataStore.CommitPolicy.OnChange,
+    )
+}
+
 class MVStoreDeleteTests :
-    AbstractDeleteTests(MVDataStore.open(DB_PATH)),
+    AbstractDeleteTests(MVDataStore.open(defaultTestConfiguration)),
     DatabaseDeleter by MVStoreDeleter
 
 class MVStoreDocumentDatabaseTests :
-    AbstractDocumentDatabaseTests(MVDataStore.open(DB_PATH)),
+    AbstractDocumentDatabaseTests(MVDataStore.open(defaultTestConfiguration)),
     DatabaseDeleter by MVStoreDeleter
 
 class MVStoreIndexTests :
-    AbstractIndexTests(MVDataStore.open(DB_PATH)),
+    AbstractIndexTests(MVDataStore.open(defaultTestConfiguration)),
     DatabaseDeleter by MVStoreDeleter
 
 class MVStoreInsertTests :
-    AbstractInsertTests(MVDataStore.open(DB_PATH)),
+    AbstractInsertTests(MVDataStore.open(defaultTestConfiguration)),
     DatabaseDeleter by MVStoreDeleter
 
 class MVStoreUpdateTests :
-    AbstractUpdateTests(MVDataStore.open(DB_PATH)),
+    AbstractUpdateTests(MVDataStore.open(defaultTestConfiguration)),
     DatabaseDeleter by MVStoreDeleter
 
 class MVStoreFindTests :
-    AbstractFindTests(MVDataStore.open(DB_PATH)),
+    AbstractFindTests(MVDataStore.open(defaultTestConfiguration)),
     DatabaseDeleter by MVStoreDeleter
 
 class MVStoreObjectCollectionTests :
-    AbstractObjectCollectionTests(MVDataStore.open(DB_PATH)),
+    AbstractObjectCollectionTests(MVDataStore.open(defaultTestConfiguration)),
     DatabaseDeleter by MVStoreDeleter
 
+class MVStorePeriodicCommitPolicyTests :
+    AbstractPeriodicCommitPolicyTests(
+        MVDataStore.open(
+            DataStore.Configuration(
+                kotlinx.io.files.Path(DB_PATH),
+                DataStore.CommitPolicy.Periodic(commitInterval),
+            ),
+        ),
+    ),
+    DatabaseDeleter by MVStoreDeleter {
+    override fun getUnsavedMemory() = (store as MVDataStore).delegate.unsavedMemory
```

Comment:
This is wrong. The tests here are meant to be agnostic of the implementation of `DataStore`. By casting it to `MVDataStore` you are imposing that his test can run only for `MVDataStore`.